File: chap11/column.py
from scipy.stats import norm, gaussian_kde

mu_all_ms = fit.draws_pd(vars=['mu_all'])
s_y_ms = fit.draws_pd(vars=['s_y']).s_y.values

t_now = 21 - 1
dens_pred = gaussian_kde(mu_all_ms.iloc[:,t_now])
w = norm.pdf(x=85.2, loc=mu_all_ms.iloc[:,t_now], scale=s_y_ms)
dens_flt = gaussian_kde(mu_all_ms.iloc[:,t_now], weights=w/sum(w))
post_mean = np.sum(mu_all_ms.iloc[:,t_now] * w/sum(w))

File: chap11/column.R
fit <- readRDS('output/result-model11-2.RDS')

mu_all_ms <- fit$draws('mu_all', format='matrix')
s_y_ms <- fit$draws('s_y', format='df')$s_y

t_now <- 21
dens_pred <- density(mu_all_ms[,t_now])
w <- dnorm(x=85.2, mean=mu_all_ms[,t_now], sd=s_y_ms)
dens_flt <- density(mu_all_ms[,t_now], weight=w/sum(w))
post_mean <- sum(mu_all_ms[,t_now] * w/sum(w))
